.github/workflows/docker.yml

```diff
@@ -13,58 +15,107 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
-permissions:
-  contents: read
-
 jobs:
 
   docker_builds:
 
     strategy:
       fail-fast: false
       matrix:
+        platform: ["arm64", "amd64"]
+        base: ["alpine-small", "alpine-normal", "ubuntu-small", "ubuntu-full"]
+        proprietary_sdks: ['false']
         include:
-          - name: alpine-small-amd64
-            image_name: alpine-small
-            arch: amd64
-
-          - name: alpine-normal-amd64
-            image_name: alpine-small
-            arch: amd64
-
-          - name: ubuntu-small-amd64
-            image_name: ubuntu-small
-            arch: amd64
-
-          - name: ubuntu-small-arm64
-            image_name: ubuntu-small
-            arch: arm64
-
-          - name: ubuntu-full-amd64-proprietary-sdks
-            image_name: ubuntu-full
-            arch: amd64
-
-          - name: ubuntu-full-arm64
-            image_name: ubuntu-full
-            arch: arm64
-
-    name: ${{ matrix.name }}
-
-    runs-on: ubuntu-latest
+          - platform: amd64
+            base: ubuntu-full
+            proprietary_sdks: true
+    name: ${{ matrix.base }}-${{ matrix.platform }} ${{ matrix.proprietary_sdks  == 'true' && 'proprietary-sdks' || '' }}
+    runs-on: ${{ matrix.platform == 'amd64' && 'ubuntu-latest' || 'ubuntu-24.04-arm64' }}
+    permissions:
+      contents: read
+      attestations: write
+      packages: write
+      id-token: write
 
     steps:
+      - name: Set PUSH_PACKAGE due to schedule
+        if: github.event.schedule=='*/0 * * * *'
```

Comment:
Checking the `event_name` will allow updating the time without having to change the rest of the workflow: https://stackoverflow.com/a/70282170

Reply:
ahh yeah that's _much_ better!